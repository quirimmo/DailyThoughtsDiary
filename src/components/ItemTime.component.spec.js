import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

const mockOpenTimePickerAndroid = jest.fn(() => ({
	action: 'ok',
	hour: 10,
	minute: 20
}));
jest.mock(
	'./../../node_modules/react-native/Libraries/Components/TimePickerAndroid/TimePickerAndroid',
	() => ({
		open: mockOpenTimePickerAndroid,
		dismissedAction: 'cancel'
	})
);

import ItemTime from './ItemTime.component';

const mockOnSetTime = jest.fn();
let component;
let instance;

describe('ItemTime', () => {
	beforeEach(() => {
		jest.useFakeTimers();
		NavigationTestUtils.resetInternalState();
		component = renderer.create(
			<ItemTime date="2018-12-25T14:00:46.723Z" onSetTime={mockOnSetTime} />
		);
		instance = component.getInstance();
	});
	afterEach(() => {
		jest.clearAllMocks();
	});

	it('should be defined', () => {
		expect(ItemTime).toBeDefined();
	});

	it('should render the component', async () => {
		expect(component.toJSON()).toMatchSnapshot();
	});

	describe('onSettingTime', () => {
		it('should call the open method of TimePickerAndroid', async () => {
			instance.onSettingTime();
			expect(mockOpenTimePickerAndroid).toHaveBeenCalledWith({
				hour: 15,
				minute: 0
			});
		});

		it('should call the onSetTime prop', async () => {
			await instance.onSettingTime();
			expect(mockOnSetTime).toHaveBeenCalledWith(10, 20);
		});
	});
});
