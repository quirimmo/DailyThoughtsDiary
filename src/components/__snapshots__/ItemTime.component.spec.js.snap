// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTime should render the component 1`] = `
Array [
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
      }
    }
  >
    Time:
  </Text>,
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        15:00
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0.5,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Change Time
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
